src/Implement/RequestHandler.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "RequestHandler.h"
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <string>
+
+std::unordered_map<std::string, APIHandler *> RequestHandler::apis;
+
+RequestHandler::RequestHandler() { this->read_size = 4096; }
+
+void RequestHandler::SetFd(const int &file_descriptor) {
+  this->client_fd = file_descriptor;
+}
+
+int RequestHandler::ReadHeader(std::string &header,
+                               std::string &msg_over_header) {
+  char input_buffer[this->read_size];
+  const std::string find_key = "Content-Length:";
+  const std::string end_of_header = "\r\n\r\n";
+  // keep read buffer until no request message
+  while (true) {
+    int msg_read = recv(this->client_fd, input_buffer, sizeof(input_buffer), 0);
+    // means end of request
+    if (msg_read <= 0) {
+      return -1;
+    }
+    // add read buffer to header
+    header += input_buffer;
```

Comment:
Default string constructor searches for a NULL (\0) character in the input char*, and that can be different from what was `recv()`'ed.

Consider explicitly constructing the string with pointer and size
`header += string(input_buffer, msg_read);`

Reply:
Done